src/librustc/lint/builtin.rs

```diff
@@ -542,7 +542,11 @@ impl LintPass for BoxPointers {
     }
 
     fn check_expr(&mut self, cx: &Context, e: &ast::Expr) {
-        let ty = ty::expr_ty(cx.tcx, e);
+        let ty = match ty::expr_ty_opt(cx.tcx, e) {
+            Some(ty) => ty,
+            None => cx.tcx.sess.span_bug(e.span, "No type for expression"),
+        };
```

Comment:
Ah right, that reminds me, I was going to fold the `span_bug` stuff into `ty::expr_ty`. Makes debugging easier.